src/server/api/creation.ts

```diff
@@ -28,13 +29,33 @@ export const creationRoutes = (cfg: APIConfig, store: Store, apsystem: ActivityP
   }, async (request, reply) => {
     const { actor } = request.params
 
+    const existedAlready = (await store.actorsDb.keys().all()).some(k => k === actor)
+
     const allowed = await apsystem.hasPermissionActorRequest(actor, request, false)
     if (!allowed) {
       return await reply.code(403).send('Not Allowed')
     }
 
     const info = request.body
     await store.forActor(actor).setInfo(info)
+
+    if (!existedAlready && info.announce) {
```

Comment:
Could you add a new class to contain all this stuff and attach it to `APSystem.announcements`? There should be a method like `apsystem.announcements.announce(actor)` which can perform activity creation.